sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEOptimizer.scala

```diff
@@ -28,7 +29,9 @@ import org.apache.spark.util.Utils
 /**
  * The optimizer for re-optimizing the logical plan used by AdaptiveSparkPlanExec.
  */
-class AQEOptimizer(conf: SQLConf) extends RuleExecutor[LogicalPlan] {
+class AQEOptimizer(session: SparkSession) extends RuleExecutor[LogicalPlan] {
```

Comment:
`SparkSession` seems an overkill. Shall we narrow down and pass over `session.sessionState.adaptiveRulesHolder` only (in addition to the existing `conf`)?